sdk/monitor/azure-monitor-opentelemetry-exporter/tests/quickpulse/test_live_metrics.py

```diff
@@ -118,3 +133,117 @@ def test_singleton(self):
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._live_metrics._get_span_document")
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._live_metrics._is_post_state")
+    def test_record_span_dep_success(self, post_state_mock, span_doc_mock, append_doc_mock):
+        post_state_mock.return_value = True
+        span_doc = mock.Mock()
+        span_doc_mock.return_value = span_doc
+        span_mock = mock.Mock()
+        span_mock.end_time = 10
+        span_mock.start_time = 5
+        span_mock.status.is_ok = True
+        span_mock.kind = SpanKind.CLIENT
+        qpm = _QuickpulseManager(
+            connection_string="InstrumentationKey=4321abcd-5678-4efa-8abc-1234567890ac;LiveEndpoint=https://eastus.livediagnostics.monitor.azure.com/",
+            resource=Resource.create(),
+        )
+        qpm._dependency_rate_counter = mock.Mock()
+        qpm._dependency_duration = mock.Mock()
+        qpm._record_span(span_mock)
+        append_doc_mock.assert_called_once_with(span_doc)
+        qpm._dependency_rate_counter.add.assert_called_once_with(1)
+        qpm._dependency_duration.record.assert_called_once_with(5 / 1e9)
+
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._live_metrics._append_quickpulse_document")
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._live_metrics._get_span_document")
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._live_metrics._is_post_state")
+    def test_record_span_dep_failure(self, post_state_mock, span_doc_mock, append_doc_mock):
+        post_state_mock.return_value = True
+        span_doc = mock.Mock()
+        span_doc_mock.return_value = span_doc
+        span_mock = mock.Mock()
+        span_mock.end_time = 10
+        span_mock.start_time = 5
+        span_mock.status.is_ok = False
+        span_mock.kind = SpanKind.CLIENT
+        qpm = _QuickpulseManager(
+            connection_string="InstrumentationKey=4321abcd-5678-4efa-8abc-1234567890ac;LiveEndpoint=https://eastus.livediagnostics.monitor.azure.com/",
+            resource=Resource.create(),
+        )
+        qpm._dependency_failure_rate_counter = mock.Mock()
+        qpm._dependency_duration = mock.Mock()
+        qpm._record_span(span_mock)
+        append_doc_mock.assert_called_once_with(span_doc)
+        qpm._dependency_failure_rate_counter.add.assert_called_once_with(1)
+        qpm._dependency_duration.record.assert_called_once_with(5 / 1e9)
+
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._live_metrics._append_quickpulse_document")
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._live_metrics._get_log_record_document")
+    @mock.patch("azure.monitor.opentelemetry.exporter._quickpulse._live_metrics._is_post_state")
+    def test_record_log_exception(self, post_state_mock, log_doc_mock, append_doc_mock):
```

Comment:
Seems like this is missing some log scenarios.

Reply:
Could you be more specific on what scenarios you are referring to?